osu.Game/Overlays/Direct/PlayButton.cs

```diff
@@ -115,6 +115,7 @@ protected override void Update()
             {
                 Playing.Value = false;
                 Preview = null;
+                trackLoader = null;

```

Comment:
have you tried not setting Preview to null as an alternative? It would mean we can reuse the preview for future clicks.

Reply:
I've tried, but it doesn't work for some reason. I'll pr this anyway, please check that when you have a chance

Reply:
Basically, AudioManager needs a rework. The issue is that `AudioCollectionManager` has its own Update tree which is completely separate from the scene graph, and removes completed tracks as an optimisation.

I believe the proper solution here for the time being is to reference the TrackManager locally and call `AddItemToList` with the track on it. In the near future we will be adding IComponents which should allow us to attach tracks to Drawables and control updates by the Drawable lifetimes.
Please add a comment describing this behaviour (e.g. "this is done because AudioAdjustmentCollection removes completed tracks").

Reply:
(This is essentially what SampleManager does to circumvent this same issue).

Reply:
why don't we just remove/fix the part where it automatically removes tracks?

Reply:
And add a member variable that can be set to disable that functionality? Idk I'm fine either way, I don't want to spend time than I already have on this isolated component.

The best solution will be IComponents in either case.